source/app/controllers/urls_controller.rb

```diff
@@ -1,2 +1,47 @@
 class UrlsController < ApplicationController
+  # GET /urls
+  # GET /urls.json
+  def index
+    redirect_to login_path and return if !logged_in?
+    @urls = Url.all
+  end
+
+  # GET /urls/1
+  # GET /urls/1.json
+  def show
+    url = Url.find_by(short_url: params[:id])
```

Comment:
I'm having trouble getting this to work. On my working branch it's throwing ActiveRecord::RecordNotFound, but no 404 seems to be happening - or at least my test is failing due to the exception. Some env setting I'm missing?